pkg/controller/node/node_controller.go

```diff
@@ -1229,8 +1279,107 @@ func (ctrl *Controller) syncMachineConfigPool(key string) error {
 			}
 			return err
 		}
+		masterTargeted = len(candidates)
 		ctrlcommon.UpdateStateMetric(ctrlcommon.MCCSubControllerState, "machine-config-controller-node", "Sync Machine Config Pool", pool.Name)
 	}
+
+	// If coordinating with arbiter pool, also handle arbiter node updates
+	if arbiterPool != nil && len(arbiterNodes) > 0 {
+		// Set cluster config annotation for arbiter nodes
+		if err := ctrl.setClusterConfigAnnotation(arbiterNodes, controlPlaneTopology); err != nil {
```

Comment:
Wondering if we could combine this with the same function call earlier somehow, or move it closer (although I guess functionally speaking it shouldn't matter)

There's quite a bit of duplicated code in general. I guess the main reason we can't just merge into the above functions is that arbiter has a different desiredConfig annotation it would need to set? Would it be easier if we modified the `updateCandidateMachines` function to account for that and treat the arbiter node as a master node in this function?

Reply:
Was playing around a bit more with trying to clean up the logic here, removed some duplication but avoided touching the `updateCandidateMachines` since we were essentially just moving the same if logic down the chain and I was a bit more cautious around that code for fear of introducing more errors.